woocommerce/woocommerce-ios
# Blaze Optimization: Sync Blaze campaigns on campaign list

Part of #10933 


## Description

This PR adds a new view for syncing and displaying Blaze campaigns for a site. Highlights:
- Campaigns are sorted by ID. This should be sufficient for listing items by created date DESC, but I will add an update to the data model to include created/end date for campaigns in a subsequent PR.
- All features for the list have been added:
  - Pagination
  - Pull-to-refresh
  - Empty state and loading state. There is no design for these, so I improvised them.
  - Campaign selection
  - Campaign creation CTA handling 
- The only entry point for now is from the Menu tab. I added a local feature flag to hide the WIP from production. ~~Ann has confirmed in a recent comment (pecCkj-13f-p2#comment-572) that we should show campaign creation if no campaign is created - this will be updated in a subsequent PR.~~
- Also in the above comment, Ann mentioned displaying spent budget while I'm displaying total budget currently. That will also need to be updated later.
- Due to the size of this PR, I'll add unit tests in a subsequent PR.

## Testing instructions

- Log in as an admin to a site that is eligible for Blaze.
- Navigate to the Menu tab.
- Select Blaze entry point.
- If your site has existing campaigns, they should be listed with the latest items first. Tapping on any item should show a web view with relevant details.
- If your site doesn't have any campaign, an empty view should be displayed.
- Tap the Create CTA, a new web view should be displayed for creating a new campaign.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/7bb3ecb2-586d-4979-8e73-e4a44e3cf5ab

Empty state:

<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/7589b8e1-0d5a-40e9-8d7c-920f17d9ad2d" width=320 />


---
